### PR TITLE
Hotfix for portal 2 crash

### PR DESCRIPTION
**Pull request by Jukspa**
Fix for issue #68

Strings in 2007 source are just const char* pointers and presumably they changed that in portal 2 so the entity property code was crashing on trying to read strings.

### CODE REVIEW

**Comment by YaLTeR** [comment]
Thanks!

File: spt/OrangeBox/modules/ClientDLL.cpp [hunk 1]
```diff
@@ -658,16 +663,23 @@ Vector ClientDLL::GetPlayerEyePos()
...
 {
+	if (!ORIG_GetLocalPlayer)
+		return false;
 	return (*reinterpret_cast<int*>(reinterpret_cast<uintptr_t>(ORIG_GetLocalPlayer()) + offFlags)) & FL_DUCKING;
 }
 
 double ClientDLL::GetDuckJumpTime()
 {
+	if (!ORIG_GetLocalPlayer)
+		return 0;
```
Comment (Margen67) [inline spt/OrangeBox/modules/ClientDLL.cpp]: Why use both `0` and `false`?

Comment (Jukspa) [reply spt/OrangeBox/modules/ClientDLL.cpp]: Don't understand the question, if you can't access the local player data, then no valid value can be returned, in which case it returns 0/false. This is just a hotfix for crashes that are caused by calling a function pointer that is null.

Comment (YaLTeR) [reply spt/OrangeBox/modules/ClientDLL.cpp]: Huh? This function returns `double` so it's `0`, the other one returns `bool` so it's `false`.

**Pull request merged**